### ISSUE TITLE
Filtered states don't change despite different fixed error variances

### ISSUE THREAD

**Issue by anapereira1982** [statsmodels/statsmodels#9464]
I generated synthetic data to verify if the filtered states change in response to different error variances.
These are the data I generated:

# Generate synthetic data
np.random.seed(42)
n = 1000
c = np.zeros(n)
y = np.zeros(n)

# AR(1) process for cycle c_t
rho1 = 0.7
for t in range(2, n):
    c[t] = rho1 * c[t-1] + np.random.normal(0, 0.5)

# Observation equation
y =  c

Then, I constructed a custom unobserved state space model. Here is the code:

# Custom state space model

class SimpleCycleModel(sm.tsa.statespace.MLEModel): 
  def __init__(self, endog, fixed_cycle_variance):

        super(SimpleCycleModel, self).__init__(endog, k_states=1,initialization="diffuse")

        self.ssm["design"] = np.array([[1.0]])
        self.ssm["transition"] = np.array([[0.8]])      # AR(1) for cycle
        self.ssm["selection"] = np.array([[1.0]])       # R_t
        self.ssm["state_cov"] = np.array([[fixed_cycle_variance]])  # Q_t

        self.fixed_cycle_variance = fixed_cycle_variance

    @property
    def param_names(self):
        return [
            "phi",               # static coeff.
            "var.cycle"         # error variance in the cycle state equation
        ]

    @property
    def start_params(self):
        # Setting the initial values of the parameters
        params = np.r_[
            0.02,  # Initial guess for the intercept of the trend state equation
            self.fixed_cycle_variance    # Initial guess
        ]

        return params

    def transform_params(self, unconstrained):
        constrained = unconstrained.copy()  # Copy of the unconstrained parameter vector

        # Ensures that AR parameters are stationary
        constrained[0:1] = constrain_stationary_univariate(constrained[0:1])

        constrained[1] = self.fixed_cycle_variance

        return constrained

    def untransform_params(self, constrained):
        unconstrained = constrained.copy()

        # Ensures that AR parameters are stationary
        unconstrained[0:1] = unconstrain_stationary_univariate(constrained[0:1])

        unconstrained[1] = self.fixed_cycle_variance

        return unconstrained

    def update(self, params, **kwargs):

        params = super(SimpleCycleModel, self).update(params, **kwargs)

        # Update AR(1) parameter
        self.ssm["transition", 0, 0] = params[0]  # rho_11 for beta cycle

        # Update state covariance matrix
        self.ssm["state_cov", 0, 0] = params[1]

# Apply the model considering different values for the error variance of the cycle component

fitted_values = np.full((n, 1), np.nan)
for fixed_variance in [1e-6, 0.1, 1.0, 10000]:
    model = SimpleCycleModel(y, fixed_variance)
    print(f"Fixed variance: {fixed_variance}")
    results = model.fit()
    print(results.summary())
    fitted_values = np.column_stack((fitted_values, results.filtered_state.T))

# Prints the filtered cycle for each value of the error variance of the cycle component
labels = ['NaN','$\\sigma$ = 1e-6', '$\\sigma$ = 0.1', '$\\sigma$ = 1.0','$\\sigma$ = 10000']
plt.figure(figsize=(10, 6))  # Set figure size
for i in range(fitted_values.shape[1]):  # Loop through columns
    plt.plot(fitted_values[:, i], label=labels[i])  # Plot each column with a label
plt.plot(y, label='Endog var')  # Plot each column with a label
plt.xlabel('Row Index')
plt.ylabel('Values')
plt.title('Plot of All Columns in fitted_values')
plt.legend()  # Show legend
plt.grid(True)  # Optional: Add a grid
plt.tight_layout()
plt.show(block=True)
plt.interactive(False)

# Results

The result is always the same filtered series. What could be the problem?

![imagem](https://github.com/user-attachments/assets/949ab4dd-0d0f-4941-b040-7b2643901606)

**Comment by tzukpolinsky** [comment]
np.random.seed(42) fixing the randomness through out all of the numpy random class.
so you are basically always run the same data and get the same results.
remove this code line:
`np.random.seed(42)
`
maybe that would solve it?

**Comment by ChadFulton** [comment]
Closing as answered, but feel free to follow up if this is unclear.
